SalesforceCommerceCloud/pwa-kit
# Fix generated project proxy config

# Description



The proxy configuration on a generated project was hardcoded with the value: `zzrf-001.sandbox.us01.dx.commercecloud.salesforce.com`.         

The changes in this PR make the proxy value dynamic, according to the selected preset or answers from the user.



# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1)

# How to Test-Drive This PR

1. Run the script using the `test-project` and verify the proxy config:
```
GENERATOR_PRESET=test-project node packages/pwa-kit-create-app/scripts/create-mobify-app.js --outputDir ./preset-test-project
```

packages/pwa/config/default.js
```
ssrParameters: {
        ssrFunctionNodeVersion: '14.x',
        proxyConfigs: [
            {
                host: 'kv7kzm78.api.commercecloud.salesforce.com',
                path: 'api'
            },
            {
                host: 'zzrf-001.sandbox.us01.dx.commercecloud.salesforce.com',
                path: 'ocapi'
            },
            {
                host: 'api.cquotient.com',
                path: 'einstein'
            }
        ]
    }
```

2. Run the script using the `retail-react-app-demo` and verify the proxy config:
``` 
GENERATOR_PRESET=retail-react-app-demo node packages/pwa-kit-create-app/scripts/create-mobify-app.js --outputDir ./preset-retail-react-app-demo3
```

packages/pwa/config/default.js
```
ssrParameters: {
        ssrFunctionNodeVersion: '14.x',
        proxyConfigs: [
            {
                host: 'kv7kzm78.api.commercecloud.salesforce.com',
                path: 'api'
            },
            {
                host: 'zzte-053.sandbox.us02.dx.commercecloud.salesforce.com',
                path: 'ocapi'
            },
            {
                host: 'api.cquotient.com',
                path: 'einstein'
            }
        ]
    }
```

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
